### PR TITLE
Compression Benchmarks for speed & size

### PR DESCRIPTION
**Pull request by fylux**
## TODO
 - Include or generate Event0.root
 - Add tests for NanoAOD

### CODE REVIEW

File: root/io/io/MainEventCompressionBenchmark.cxx [hunk 1]
```diff
@@ -0,0 +1,119 @@
+#include "TFile.h"
+#include "TTree.h"
+
+#include "benchmark/benchmark.h"
+#include "rootbench/RBConfig.h"
+
+static std::string GetAlgoName(int algo) {
+   std::string algo_name = "";
+   if (algo == 0) {
```
Comment (oshadura) [inline root/io/io/MainEventCompressionBenchmark.cxx]: 0 means actually default compression algorithm (which is zlib in this case). For zlib it is 1. Can you add please  "algo ==1" as a conditional statements? I think here you can use switch statement easily.

File: root/io/io/MainEventCompressionBenchmark.cxx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+static void BM_MainEvent_Compress_LZMA(benchmark::State &state) {
+   BM_MainEvent_Compress(state, 2);
+}
+static void BM_MainEvent_Compress_LZ4(benchmark::State &state) {
+   BM_MainEvent_Compress(state, 4);
+}
+
+
+static void BM_MainEvent_Decompress_ZLIB(benchmark::State &state) {
+   BM_MainEvent_Decompress(state, 0);
```
Comment (oshadura) [inline root/io/io/MainEventCompressionBenchmark.cxx]: Same here ;) ZLIB uses "1" as a code for compression algorithm settings.

File: root/io/io/MainEventCompressionBenchmark.cxx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      Int_t ev;
+
+      for (ev = 0; ev < nevent; ev++) {
+         nb += tree->GetEntry(ev);
+      }
+   }
+}
+
+static void BM_MainEvent_Compress_ZLIB(benchmark::State &state) {
+   BM_MainEvent_Compress(state, 0);
```
Comment (oshadura) [inline root/io/io/MainEventCompressionBenchmark.cxx]: Same here ;) ZLIB uses "1" as a code for compression algorithm settings.

File: root/io/io/CMakeLists.txt [hunk 4]
```diff
@@ -6,4 +6,9 @@ RB_ADD_GBENCHMARK(IOBenchmarks
 RB_ADD_GBENCHMARK(CompressBenchmarks
   TFileBenchmarks.cxx
   LABEL short
+  LIBRARIES Core RIO ROOTDataFrame MathCore) 
+
+RB_ADD_GBENCHMARK(CompressBenchmarks2
```
Comment (oshadura) [inline root/io/io/CMakeLists.txt]: Maybe not  CompressBenchmarks2, but TFile_CompressionBenchmarks?

Comment (oshadura) [reply root/io/io/CMakeLists.txt]: I will also probably rename CompressBenchmarks as TFile_RDFSnapshot to make more clear what are you testing..

File: root/io/io/CMakeLists.txt [hunk 5]
```diff
@@ -6,4 +6,9 @@ RB_ADD_GBENCHMARK(IOBenchmarks
...
+  MainEventCompressionBenchmark.cxx
+  LABEL short
   LIBRARIES Core RIO ROOTDataFrame MathCore)
```
Comment (oshadura) [inline root/io/io/CMakeLists.txt]: I think here you don't need ROOTDataFrame MathCore as a dependencies.

File: root/io/io/MainEventCompressionBenchmark.cxx [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+   else if (algo == 4) {
+      algo_name = "lz4";
+   } 
+   else if (algo == 5) {
+      algo_name = "zstd";
+   }
+   return algo_name;
+}
+
+static void BM_MainEvent_Compress(benchmark::State &state, int algo) {
```
Comment (oshadura) [inline root/io/io/MainEventCompressionBenchmark.cxx]: We will need to rename benchmarks here, please check https://github.com/root-project/rootbench#simple-benchmark-template

File: root/io/io/CMakeLists.txt (hunk 4 above)
Comment (oshadura) [reply root/io/io/CMakeLists.txt]: More info about coding conventions are here: https://github.com/root-project/rootbench#simple-benchmark-template

File: root/io/io/MainEventCompressionBenchmark.cxx (hunk 6 above)
Comment (fylux) [reply root/io/io/MainEventCompressionBenchmark.cxx]: Could you please provide an example of what name should it have?

Comment (oshadura) [reply root/io/io/MainEventCompressionBenchmark.cxx]: Maybe BM_TFile_MainEventCompress? (should be BM_CLASSNAME_ROUTINE)

**Comment by oshadura** [comment]
@fylux here you will need to rebase please! (I will send you patch for Event.root generation as soon as possible)

**Comment by fylux** [comment]
> 
> 
> @fylux here you will need to rebase please! (I will send you patch for Event.root generation as soon as possible)

Ok

**Comment by oshadura** [comment]
I merged benchmarks in other PR :)

**Pull request closed without merging**